storm-core/src/clj/org/apache/storm/daemon/worker.clj

```diff
@@ -577,11 +577,11 @@
 (defserverfn mk-worker [conf shared-mq-context storm-id assignment-id port worker-id]
   (log-message "Launching worker for " storm-id " on " assignment-id ":" port " with id " worker-id
                " and conf " conf)
-  (if-not (ConfigUtils/isLocalMode conf)
-    (redirect-stdio-to-slf4j!))
   ;; because in local mode, its not a separate
   ;; process. supervisor will register it in this case
-  (when (= :distributed (ConfigUtils/clusterMode conf))
+  ;; if (ConfigUtils/isLocalMode conf) returns false then it is in distributed mode.
+  (when-not (ConfigUtils/isLocalMode conf)
```

Comment:
https://issues.apache.org/jira/browse/STORM-1516?focusedCommentId=15134067&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-15134067

Satish already explains why it occurs via JIRA comment.